File: python/code.py
import hashlib
import hmac
import base64

def generate_signature(algorithm, key, data):
    hashed = hmac.new(key.encode('utf-8'), data.encode('utf-8'), hashlib.__dict__[algorithm])
    signature = base64.b64encode(hashed.digest()).decode('utf-8')
    return signature

algorithm = 'sha256'
key = 'your-secret-key'
data = 'data-to-sign'
signature = generate_signature(algorithm, key, data)
print("Signature:", signature)

import secrets
nonce = secrets.token_hex(16)
print("Nonce:", nonce)

File: php/code.php
function generateSignature($algorithm, $key, $data) {
    $signature = hash_hmac($algorithm, $data, $key, true);
    return base64_encode($signature);
}

$algorithm = 'sha256';
$key = 'your-secret-key';
$data = 'data-to-sign';
$signature = generateSignature($algorithm, $key, $data);
echo "Signature: " . $signature . "\n";

$nonce = bin2hex(random_bytes(8));
echo "Nonce: " . $nonce . "\n";
